test/parallel/test-stream-readable-to-web.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+require('../common');
+const { Readable } = require('stream');
+const process = require('process');
+const fs = require('fs');
+const assert = require('assert');
+
+// Based on: https://github.com/nodejs/node/issues/46347#issuecomment-1413886707
+{
+  let currentMemoryUsage = process.memoryUsage().arrayBuffers;
+
+  // We initialize a stream, but not start consuming it
+  const randomNodeStream = fs.createReadStream('/dev/urandom');
+  // after 2 seconds, it'll get converted to web stream
+  let randomWebStream;
+
+  // We check memory usage every second
+  // since it's a stream, it shouldn't be higher than the chunk size
+  const reportMemoryUsage = () => {
+    const { arrayBuffers } = process.memoryUsage();
+    currentMemoryUsage = arrayBuffers;
+
+    assert(currentMemoryUsage <= 256 * 1024 * 1024);
+  };
+  setInterval(reportMemoryUsage, 1000);
```

Comment:
I wonder if this test could be somewhat flakey? too many timeouts but nonetheless shouldn't be a blocker